enhancements/update/object-removal-manifest-annotation.md

```diff
@@ -0,0 +1,228 @@
+* `0000_50_cluster-svcat-apiserver-operator_06_service.yaml` containing a Service in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the Service would be removed as part of the namespace deletion.
+* `0000_50_cluster-svcat-apiserver-operator_07_deployment.yaml` containing a Deployment in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the Deployment would be removed as part of the namespace deletion.
+* `0000_50_cluster-svcat-apiserver-operator_08_cluster-operator.yaml` containing a cluster-scoped ClusterOperator.
+    The deletion annotation would be added to this manifest.
+* `0000_90_cluster-svcat-apiserver-operator_00_prometheusrole.yaml` containing a Role in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the Role would be removed as part of the namespace deletion.
+* `0000_90_cluster-svcat-apiserver-operator_01_prometheusrolebinding.yaml` containing a RoleBinding in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the RoleBinding would be removed as part of the namespace deletion.
+* `0000_90_cluster-svcat-apiserver-operator_02-operator-servicemonitor.yaml` containing a ServiceMonitor in the `openshift-service-catalog-apiserver-operator` namespace.
+    This manifest would be dropped, because the ServiceMonitor would be removed as part of the namespace deletion.
+
+So the remaining manifests with deletion annotations would be the namespace and the cluster-scoped CRD, ServiceCatalogAPIServer, ClusterRoleBinding, and ClusterOperator.
+The ordering of the surviving manifests would not be particularly important, although keeping the namespace first to avoid removing the ClusterRoleBinding while the consuming Deployment was still running.
+Although in the event of racing deletions, it's hard to see how a Deployment whose ClusterRoleBinding had been removed could get up to much trouble.
+However, in situations like this where multiple deletions are required, it is up to the developer to name the manifests such that deletions occur in the correct order.
+
+Similar handling would be required for the svcat-controller-manager operator.
+
+If resources external to kubernetes must be removed the developer must provide the means to do so.
+This is expected to be done through modification of an operator to do the removals during it's finalization.
+If operator modification for object removal is necessary that operator would be deleted in a subsequent update.
+This eliminates the need for new, possibly complex, CVO logic to handle both an update and a delete of the same object.
+
+If this enhancement had been implemented in 4.5 with the deletion manifests proposed above, the deletion manifests would have been preserved through 4.5.z and removed in 4.6.
```

Comment:
This has talked about tagging these manifests, but what about the overall flow?

1. in 4.4, the service catalog operator behaves normally
2. in 4.5, the service catalog operator does something (you didn't specify what) to get ready for CVO deletion in 4.6.  Maybe it blocks upgrades until it is in a state where it is safe for removal (this seems like a requirement).  This is where the actual work happens and the ugpradeable conditions need to be planned, blocking, and used.
3. in 4.6, the service catalog marks its CVO manifests for deletion
4. in 4.7, the service catalog removes itself from the payload.

Reply:
We can add an overall flow.

Discussed #2 with @wking and we couldn't come up with any operator specific logic that would be outside of the operator developer's responsibility so would not expect to cite it here. We're not aware of any for the svcat example.

Reply:
> Discussed #2 with @wking and we couldn't come up with any operator specific logic that would be outside of the operator developer's responsibility so would not expect to cite it here. We're not aware of any for the svcat example.

For the service catalog, if you remove the operator by deleting the manifest, what removes the service catalog API and controller?  I agree that it will be particular for each operator, but providing an example would be very helpful.

It's not clear what to do in the orphan case (a time bomb for service catalog apiserver because of certs) or in the removal case (preventing a user from progressing to the next version until the thing being managed is gone).  While you don't own that code, thinking about it and describing how an operator actually uses this to achieve both of those goals (one at a time obviously), will be helpful to them and helpful to you when you help them.

Reply:
> For the service catalog, if you remove the operator by deleting the manifest, what removes the service catalog API and controller? I agree that it will be particular for each operator, but providing an example would be very helpful.

In our example, which is specific to service catalog, the operands get deleted with their namespaces. We should explicitly state that, so will make that change. 

> It's not clear what to do in the orphan case (a time bomb for service catalog apiserver because of certs) or in the removal case (preventing a user from progressing to the next version until the thing being managed is gone). While you don't own that code, thinking about it and describing how an operator actually uses this to achieve both of those goals (one at a time obviously), will be helpful to them and helpful to you when you help them.

So I think the overall point to make is that the operator should never allow itself to be removed if it's operand is still around. Even in the case of our service catalog example where the operand is not explicitly removed by the operator the operator could still check and ensure it's operand has been removed. If it has not the operator could report that and/or attempt to do an explicit delete. I'm not sure what else to write regarding this topic since ultimately it's the developers job to ensure the removal works by thoroughly testing.

Reply:
Added additional user story per slack conversation and additional guidance details in user stories introductory paragraph.